src/charm.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, *args):
         )
 
         self.framework.observe(self.on.kratos_pebble_ready, self._on_pebble_ready)
-        self.framework.observe(self.database.on.database_created, self._on_database_changed)
+        self.framework.observe(self.database.on.database_created, self._on_database_created)
         self.framework.observe(self.database.on.endpoints_changed, self._on_database_changed)
```

Comment:
I know it was me who suggested observing `endpoints_changed`, but after giving it a second thought, we may not need it. I initially thought the endpoint would be an ip and thus could change, but it's referencing the postgresql-k8s-primary's svc name, so it will always remain `postgresql-k8s-primary.namespace.svc.cluster.local:5432`. What do you think?